packages/typespec-client-generator-core/test/package.test.ts

```diff
@@ -2176,6 +2178,41 @@ describe("typespec-client-generator-core: package", () => {
       strictEqual(apiVersionOpParam.correspondingMethodParams[0], apiVersionClientParam);
     });
 
+    it("global parameters", async () => {
+      const runnerWithArm = await createSdkTestRunner({
+        librariesToAdd: [AzureResourceManagerTestLibrary, AzureCoreTestLibrary, OpenAPITestLibrary],
+        autoUsings: ["Azure.ResourceManager", "Azure.Core"],
+        emitterName: "@azure-tools/typespec-java",
+      });
+      await runnerWithArm.compileWithBuiltInAzureResourceManagerService(`
+        model MyProperties {
+          @visibility("read")
+          @doc("Display name of the Azure Extended Zone.")
+          displayName: string;
+        }
+
+        @subscriptionResource
+        model MyModel is ProxyResource<MyProperties> {
+          @key("extendedZoneName")
+          @segment("extendedZones")
+          @path
+          name: string;
+        }
+
+        @armResourceOperations
+        interface MyInterface {
+          get is ArmResourceRead<MyModel>;
+        }
+      `);
+
+      const sdkPackage = runnerWithArm.context.sdkPackage;
+      const client = sdkPackage.clients[0].methods.find((x) => x.kind === "clientaccessor")
+        ?.response as SdkClientType<SdkHttpOperation>;
+      for (const name of ["apiVersion", "subscriptionId", "endpoint", "credential"]) {
+        ok(client.initialization.properties.find((x) => x.name === name) !== undefined);
```

Comment:
double confirm, those parameters would appear in `SdkHttpOperation.parameters` with `onClient === true`, right?

Reply:
Yeah

Reply:
Yeah, I added test for `onClient`